ui/src/message_popup/mobile_prototype_card.jsx

```diff
@@ -0,0 +1,239 @@
+            return (
+              <Message text={response} question={this.state.question} type='user' />
+            );
+          })}
+        </div>
+        <div style={styles.responseContainer}>
+          <Divider />
+          <div style={styles.responseTextBox}>
+            <TextField
+              id="responseTextField"
+              multiLine={true}
+              hintText="Type something you would say"
+              textareaStyle={styles.responseTextAreaInner}
+              rows={2}
+              underlineShow={false}
+              rowsMax={2}
+              disabled={this.state.responses.length > 10}
+              value={this.state.currentResponse}
+              onChange={this.onTextChange}
+              style={{width: '100%', paddingLeft: 20, paddingRight: 20}}
+            />
+          </div>
+          <div style={styles.responseButtonRow}>
+            <RaisedButton 
+              label={this.state.responses.length > 10 ? 'Next' : 'Send'}
+              secondary={true}
+              onTouchTap={this.state.responses.length > 10 ? this.next : this.sendMessage}
+            />
+          </div>
+        </div>
+        
+        <Dialog
+          title="Student Information"
+          actions={studentDialogActions}
+          modal={false}
+          open={this.state.currentStudent !== undefined}
+          style={{backgroundColor: 'rgba(0, 0, 0, 0.1)'}}
+          onRequestClose={this.closeStudentDialog}
+        >
+          <div style={styles.studentName}>{name}</div>
+          <div style={styles.studentAttribute}>{`${grade} ${gender}, ${race}`}</div>
+          {behavior && <div style={styles.studentAttribute}>{behavior}</div>}
+          {ell && <div style={styles.studentAttribute}>{ell}</div>}
+          {learningDisabilities && <div style={styles.studentAttribute}>{learningDisabilities}</div>}
+          {academicPerformance && <div style={styles.studentAttribute}>{academicPerformance}</div>}
+          {interests && <div style={styles.studentAttribute}>{interests}</div>}
+          {familyBackground && <div style={styles.studentAttribute}>{familyBackground}</div>}
+          {ses && <div style={styles.studentAttribute}>{ses}</div>}
+        </Dialog>
+        
+      </div>
+    );
+  },
+});
+
+const Message = React.createClass({
+  propTypes: {
+    text: React.PropTypes.string.isRequired,
+    question: React.PropTypes.object.isRequired,
+    type: React.PropTypes.string.isRequired,
+    openStudentDialog: React.PropTypes.func
+  },
+  
+  componentDidUpdate(){ 
+    ReactDom.findDOMNode(this).scrollIntoView(); 
+  },
+  
+  render(){
+    const questionStyle = this.props.type==='user' ? _.merge({justifyContent: 'flex-end'}, styles.question) : _.merge({justifyContent: 'flex-start'}, styles.question);
+    
+    const questionTextStyle = this.props.type==='user' ? _.merge({backgroundColor: '#e6f9ff'}, styles.questionText) : styles.questionText;
+    
+    return (
```

Comment:
I think it might be more direct to split these into two or three small components, maybe `MessageFromStudent`, `InfoMessage` and `MessageFromUser` or something like that?  Each of those can reuse styles like you're doing now.